client/parser/examples_test.go

```diff
@@ -25,32 +20,32 @@ func TestExamples(t *testing.T) {
 
 	examplesDir := filepath.Join(filepath.Dir(filepath.Dir(dir)), "examples")
 
-	err = filepath.Walk(examplesDir,
+	filepath.Walk(examplesDir,
 		func(path string, info os.FileInfo, err error) error {
+			label := fmt.Sprintf("examples test for %s", path)
 			if err != nil {
+				t.Error(label)
+				t.Errorf("filepath walk error %s", err)
 				return err
 			}
 
 			if info.IsDir() {
 				return nil
 			}
 
-			fmt.Printf("● %s\n", path)
-
-			ast, err := ParseFile(path)
+			contents, err := ioutil.ReadFile(path)
 			if err != nil {
+				t.Error(label)
+				t.Error("issue reading file contents")
 				return err
 			}
 
-			scoreUpdates, err := ast.Updates()
-			if err != nil {
-				return err
-			}
+			executeParseTestCases(t, parseTestCase{
+				label: label,
+				given: string(contents),
+			})
 
-			score := model.NewScore()
-			return score.Update(scoreUpdates...)
-		})
-	if err != nil {
-		t.Errorf("%v\n", err)
-	}
```

Comment:
I think this part might be important, although I'm not 100% sure. I think without it, the `filepath.Walk(...)` expression could be returning an error, and we're just ignoring it.

Reply:
...although, maybe `t.Error` and `t.Errorf` have the side effect of making the test case fail, so effectively it doesn't matter?

Just to be safe, could you test making one of the example files deliberately wrong and ensuring that the test case fails?

Reply:
Good point. It looks like `t.Error` does make the test fail (without returning error). I think we `return err` within the walk more as a short-circuit (e.x. no point running test case if can't read file).

Resolving.